src/test.cc

```diff
@@ -75,10 +75,12 @@ char* mkdtemp(char* name_template) {
 
 string GetSystemTempDir() {
 #ifdef _WIN32
-  char buf[1024];
-  if (!GetTempPath(sizeof(buf), buf))
+  TCHAR buf[MAX_PATH];
+  // GetTempPath returns zero if no path is found
+  if(GetTempPath(MAX_PATH, buf) == 0)
```

Comment:
You're not actually calling a different function, just changing the buffer size, and using TCHAR (which is...?). Why is there a need to call `WideToUTF8`? 

If TCHAR is `wchar_t`, how is this version of the function any better than the `char` version? Wouldn't the Windows API do the conversion for us in exactly the same way, if not better, than what `WideToUTF8()` does?